### PR TITLE
LPS-103759 | LPS-103891 Update tests to cover compound components structures and route params passing

### PR DESCRIPTION
**Pull request by albinojunior**

### CODE REVIEW

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePage.es.js [hunk 1]
```diff
@@ -9,52 +9,69 @@
...
+	);
+
+	const keywords = search.length ? search : null;
 
-		return client.get(requestUrl, {params}).then(({data}) => {
-			setData(data);
-		});
-	};
+	const selectedFilters = getSelectedItems(filterResults);
+	const hasFilters = search.length > 0 || selectedFilters.length > 0;
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePage.es.js]: use the singular form to express boolean variables

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePage.es.js [hunk 2]
```diff
@@ -9,52 +9,69 @@
...
+		<PromisesResolver promises={promises}>
+			<PerformanceByAssigneePage.Header
+				dispatch={dispatch}
+				routeParams={{...routeParams, search: keywords}}
+				selectedFilters={selectedFilters}
+				totalCount={data.totalCount}
+			/>
+			<PerformanceByAssigneePage.Body
+				data={data}
+				hasFilters={hasFilters}
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePage.es.js]: [https://github.com/gabrielpramos/liferay-portal/pull/13/files#r348718114](url)

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePageBody.es.js [hunk 3]
```diff
@@ -41,35 +41,30 @@ const Body = ({data}) => {
...
-		<EmptyState
-			className="border-0"
-			data-testid="emptyState"
-			hideAnimation={true}
-			message={Liferay.Language.get('there-is-no-data-at-the-moment')}
-			messageClassName="small"
-		/>
-	);
+const EmptyView = ({hasFilters}) => {
+	const emptyMessage = hasFilters
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePageBody.es.js]: Apply rename to the boolean variable parameters as well

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePageBody.es.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ import LoadingState from '../../shared/components/loading/LoadingState.es';
 import PromisesResolver from '../../shared/components/request/PromisesResolver.es';
 import {Table} from './PerformanceByAssigneePageTable.es';
 
-const Body = ({data}) => {
+const Body = ({data, hasFilters}) => {
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePageBody.es.js]: Apply rename to the boolean variable parameters as well

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js [hunk 5]
```diff
@@ -11,176 +11,73 @@
...
-} from './WorkloadByAssigneePageBody.es';
-import {Item, Table} from './WorkloadByAssigneePageTable.es';
+import {Body} from './WorkloadByAssigneePageBody.es';
+import {Header} from './WorkloadByAssigneePageHeader.es';
 
-const WorkloadByAssigneePage = ({page, pageSize, processId, search, sort}) => {
-	useProcessTitle(processId, Liferay.Language.get('workload-by-assignee'));
+const WorkloadByAssigneePage = ({query, routeParams}) => {
+	const {search = ''} = parse(query);
+	const {processId} = routeParams;
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js]: Sort

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js [hunk 6]
```diff
@@ -11,176 +11,73 @@
...
 	const {roleIds, taskKeys} = useFilterItemKeys(filterKeys, filterValues);
 
-	let keywords;
+	const filterResults = getFilterResults(
+		filterKeys,
+		filterTitles,
+		filterValues
+	);
+
+	const keywords = search.length ? search : null;
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js]: Sort

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js [hunk 7]
```diff
@@ -11,176 +11,73 @@
...
 
-	if (typeof search === 'string' && search) {
-		keywords = decodeURIComponent(search);
-	}
+	const selectedFilters = getSelectedItems(filterResults);
+	const hasFilters = search.length > 0 || selectedFilters.length > 0;
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js]: Sort when "hasFilters" get fixed

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/search-field/SearchField.es.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+		location,
+		match: {params, path}
+	} = useRouter();
+	const query = parse(location.search);
+	const {search = ''} = query;
+
+	const {disabled, placeholder = Liferay.Language.get('search-for')} = props;
+
+	const [searchValue, setSearchValue] = useState('');
+	const [redirect, setRedirect] = useState(false);
```
Comment (gabrielpramos) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/search-field/SearchField.es.js]: Sort

**Pull request closed without merging**